packages/flutter/lib/src/push/parse_push.dart

```diff
@@ -0,0 +1,74 @@
+part of flutter_parse_sdk_flutter;
+
+class ParsePush {
+  static final ParsePush instance = ParsePush._internal();
+  static String keyType = "gcm";
+  static String keyPushType = 'pushType';
+
+  factory ParsePush() {
+    return instance;
+  }
+
+  ParsePush._internal();
+
+  Future<void> initialize(firebaseMessaging) async {
+    // Get Google Cloud Messaging (GCM) token
+    firebaseMessaging.getToken().asStream().listen((event) async {
+      // Set token in installation
+      sdk.ParseInstallation parseInstallation =
+          await sdk.ParseInstallation.currentInstallation();
+
+      parseInstallation.deviceToken = event;
+      parseInstallation.set(keyPushType, keyType);
+
+      await parseInstallation.save();
+    });
+  }
+
+  void onMessage(message) {
+    String pushId = message.data["push_id"] ?? "";
+    String timestamp = message.data["time"] ?? "";
+    String dataString = message.data["data"] ?? "";
+    String channel = message.data["channel"] ?? "";
+
+    Map<String, dynamic>? data;
+    try {
+      data = json.decode(dataString);
+    } catch (_) {}
+
+    _handlePush(pushId, timestamp, channel, data);
+  }
+
+  void _handlePush(String pushId, String timestamp, String channel,
+      Map<String, dynamic>? data) {
+    if (pushId.isEmpty || timestamp.isEmpty) {
+      return;
+    }
+
+    if (data != null) {
+      // Show push notification
+      ParseNotification.instance.showNotification(data["alert"]);
+    }
+  }
+
+  Future<void> subscribeToChannel(String value) async {
+    sdk.ParseInstallation parseInstallation =
+        await sdk.ParseInstallation.currentInstallation();
+
+    await parseInstallation.subscribeToChannel(value);
+  }
+
+  Future<void> unsubscribeFromChannel(String value) async {
+    sdk.ParseInstallation parseInstallation =
+        await sdk.ParseInstallation.currentInstallation();
+
+    await parseInstallation.unsubscribeFromChannel(value);
+  }
+
+  Future<List<dynamic>> getSubscribedChannels() async {
+    sdk.ParseInstallation parseInstallation =
+        await sdk.ParseInstallation.currentInstallation();
```

Comment:
These seem to be really short line lengths; which makes this difficult to read. What line length have we set in the linter?

Reply:
80 char per line is the defalut. In the Flutter ecosystem, this is the standard line length.

Reply:
Seems outdated in the age of widescreen monitors, but fine.